### PR TITLE
extend and improve Mempool tests

### PR DESCRIPTION
**Pull request by arvidn**
(please don't squash these commits).

These commits are suitable to be reviewed independently, one at a time.

### CODE REVIEW

File: tests/core/full_node/test_mempool.py [hunk 1]
```diff
@@ -95,6 +101,29 @@ async def test_basic_mempool(self, two_nodes):
         assert spend_bundle is not None
 
 
+@peer_required
+@api_request
+@bytes_required
+async def respond_transaction(
```
Comment (arvidn) [inline tests/core/full_node/test_mempool.py:107]: this is "inspired" by `FullNodeAPI.respond_transaction()`, but it preserves the return values from `FullNode.respondTransaction()`, to allow tests to assert they are what we expect them to be

**Review by Yostra — APPROVED** [review]
Nice

**Comment by Yostra** [comment]
I assume CI failure is unrelated? I don't see any change that could have caused it

**Comment by arvidn** [comment]
oh, I had missed the CI failure. It's not red, as if it was cancelled or infrastructure failure. I'll look into it

**Comment by hoffmang9** [comment]
I see no CI failure so I'm confused?

**Comment by arvidn** [comment]
I re-triggered the test and it passed.

**Pull request merged**